bin/cylc-test-battery

```diff
@@ -91,10 +91,28 @@ Run only tests under "tests/cyclers/", and skip 00-daily.t
   cylc test-battery tests/cyclers
 Run the first quarter of the test battery
   cylc test-battery --chunk '1/4'
+Re-run failed tests
+  cylc test-battery --state=save
+  cylc test-battery --state=failed
 eof
 }
 
-TESTS=""
+chunk () {
+    # argument in the format chunk_no/no_chunks
+    IFS=$'/' read CHUNK_NO CHUNKS <<< "$1"
+    # create lists of tests in a tempdir
+    TMPDIR="$(mktemp -d)"
```

Comment:
Are we actually meant to override `TMPDIR` here?

Reply:
Git blame implicates your recent change!

Reply:
I see we use `${TMPDIR:-/tmp}` in the test header.

Reply:
Just wondering if there was a reason for this... `$TMPDIR` is not guaranteed to be defined is it? And `mdtemp -d` uses `$TMPDIR` if it is defined?

Reply:
Tempdir is [not defined](https://github.com/metomi/rose/pull/2222/files#diff-bb7d523d7517068df0a95d8c1bcea452R27) in the Travis-CI environment but there was no need to define it in `cylc test-battery` like this (we only needed a tempfile anyway).

The chunk test [now passes](https://travis-ci.org/cylc/cylc/jobs/419542732#L1149) with this changed.